### PR TITLE
Feature/forgot password email

### PR DESCRIPTION
**Pull request by mariaruth1**
added expiration to the tokens for the email links

### CODE REVIEW

File: IntegrationTests/Tests/AccountTests.cs [hunk 1]
```diff
@@ -13,54 +14,54 @@ public class AccountTests(VitalApiFactory vaf) : TestBase(vaf)
...
 
         var response =
-            await _client.PostAsync("/Identity/Account/Forgot-Password", JsonContent.Create(forgotPassordDto));
+            await _client.PostAsync("/Identity/Account/Forgot-Password", JsonContent.Create(forgotPasswordDto));
 
         response.StatusCode.Should().Be(HttpStatusCode.OK);
     }
 
     [Fact]
     public async Task Forgot_Password_return_500()
```
Comment (juuwel) [inline IntegrationTests/Tests/AccountTests.cs:45]: why is this called return_500 if the assertion is for OK?

File: IntegrationTests/Tests/AccountTests.cs [hunk 2]
```diff
@@ -288,4 +292,86 @@ public async Task Verify_email_return_200()
...
+
+    [Fact]
+    public async Task Change_Password_Valid_Old_Password_Changes_Password()
+    {
+        await ClearToken();
+        // Arrange
+        var newUser = new ApplicationUser()
+        {
+            Email = "change-password@app.com",
+            UserName = "change-password@app.com",
```
Comment (juuwel) [inline IntegrationTests/Tests/AccountTests.cs:319]: i think it is sufficient to only specify email, but this is ofc not wrong either

File: Vital-Frontend/src/app/toasts/toast.component.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ import {animate, style, transition, trigger} from "@angular/animations";
            max-w-[300px] w-[200px] z-50 min-h-[100px] max-h-[150px]
            overflow-ellipsis
            lg:w-[300px]"
-           [class.bg-secondary]="toast.type === 'success'" [class.bg-error]="toast.type === 'error'"
+           [class.bg-success]="toast.type === 'success'" [class.bg-error]="toast.type === 'error'"
```
Comment (juuwel) [inline Vital-Frontend/src/app/toasts/toast.component.ts:14]: the success colour feels very disruptive to our very harmonic theme lmao

File: Vital/Controllers/AccountController.cs [hunk 4]
```diff
@@ -63,16 +65,53 @@ public async Task<IActionResult> ResetPassword([FromBody] ResetPasswordDto dto)
 
         var user = await _userManager.FindByIdAsync(dto.UserId.ToString());
 
-        if (user is not null)
+        if (user is null)
+        {
+            throw new ResetPasswordException("Something went wrong.");
```
Comment (juuwel) [inline Vital/Controllers/AccountController.cs:70]: i think we could just return here, instead of throwing an exception

**Review by juuwel — APPROVED** [review]
nice job

File: IntegrationTests/Tests/AccountTests.cs (hunk 1 above)
Comment (mariaruth1) [reply IntegrationTests/Tests/AccountTests.cs:45]: ask mr Mazur, i will change it though

File: Vital-Frontend/src/app/toasts/toast.component.ts (hunk 3 above)
Comment (mariaruth1) [reply Vital-Frontend/src/app/toasts/toast.component.ts:14]: i have no strong opinion, i will change it

File: IntegrationTests/Tests/AccountTests.cs (hunk 2 above)
Comment (juuwel) [reply IntegrationTests/Tests/AccountTests.cs:319]: lmao

File: Vital/Controllers/AccountController.cs (hunk 4 above)
Comment (mariaruth1) [reply Vital/Controllers/AccountController.cs:70]: i don't think i get it

**Pull request merged**